test/sync_2dev_test.go

```diff
@@ -0,0 +1,159 @@
+	dstDir := t.TempDir()
+
+	// Spin up two instances to sync the data.
+	testSyncTwoDevicesFolders(t, srcDir, dstDir)
+
+	// Check that the destination folder now contains the same files as the source folder.
+	compareTrees(t, srcDir, dstDir)
+}
+
+func TestSyncMergeTwoDevices(t *testing.T) {
+	t.Parallel()
+
+	// Create a source folder with some data in it.
+	srcDir := generateTree(t, 50)
+	// Create a destination folder that also has some data in it.
+	dstDir := generateTree(t, 50)
+
+	// Spin up two instances to sync the data.
+	testSyncTwoDevicesFolders(t, srcDir, dstDir)
+
+	// Check that both folders are the same, and the file count should be
+	// the sum of the two.
+	if total := compareTrees(t, srcDir, dstDir); total != 100 {
+		t.Fatalf("expected 100 files, got %d", total)
+	}
+}
+
+func testSyncTwoDevicesFolders(t *testing.T, srcDir, dstDir string) {
+	t.Helper()
+
+	// The folder needs an ID.
+	folderID := rand.String(8)
+
+	// Start the source device.
+	src := startInstance(t)
+	srcAPI := rc.NewAPI(src.apiAddress, src.apiKey)
+
+	// Start the destination device.
+	dst := startInstance(t)
+	dstAPI := rc.NewAPI(dst.apiAddress, dst.apiKey)
+
+	// Add the peer device to each device. Hard code the sync addresses to
+	// speed things up.
+	if err := srcAPI.Post("/rest/config/devices", &config.DeviceConfiguration{
+		DeviceID:  dst.deviceID,
+		Addresses: []string{fmt.Sprintf("tcp://127.0.0.1:%d", dst.tcpPort)},
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+	if err := dstAPI.Post("/rest/config/devices", &config.DeviceConfiguration{
+		DeviceID:  src.deviceID,
+		Addresses: []string{fmt.Sprintf("tcp://127.0.0.1:%d", src.tcpPort)},
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// Add the folder to both devices.
+	if err := srcAPI.Post("/rest/config/folders", &config.FolderConfiguration{
+		ID:             folderID,
+		Path:           srcDir,
+		FilesystemType: fs.FilesystemTypeBasic,
+		Type:           config.FolderTypeSendReceive,
+		Devices: []config.FolderDeviceConfiguration{
+			{DeviceID: src.deviceID},
+			{DeviceID: dst.deviceID},
+		},
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+	if err := dstAPI.Post("/rest/config/folders", &config.FolderConfiguration{
+		ID:             folderID,
+		Path:           dstDir,
+		FilesystemType: fs.FilesystemTypeBasic,
+		Type:           config.FolderTypeSendReceive,
+		Devices: []config.FolderDeviceConfiguration{
+			{DeviceID: src.deviceID},
+			{DeviceID: dst.deviceID},
+		},
+	}, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// Listen to events; we want to know when the folder is fully synced. We
+	// consider the other side in sync when we've received an index update
+	// from them and subsequently a completion event with percentage equal
+	// to 100. At that point they should be done. Wait for both sides to do
+	// their thing.
+
+	waitForSync := func(api *rc.API, done chan struct{}) {
+		defer close(done)
+		lastEventID := 0
+		remoteCompletion := 0.0
+		remoteIndexUpdated := false
+	loop:
```

Comment:
If you're only using this label for `break loop` then you can lose the label and just `return`, since nothing happens after the loop.